### PR TITLE
[OSDOCS-3599] OSD Restructure

### PR DESCRIPTION
**Pull request by EricPonvelle**
Version(s):
Enterprise-4.9+

Issue:
[OSDOCS-3599](https://issues.redhat.com/browse/OSDOCS-3599)

Link to docs preview:
[Before restructure](https://docs.openshift.com/dedicated/welcome/index.html)

[After restructure](https://deploy-preview-45428--osdocs.netlify.app/openshift-dedicated/latest/welcome/index.html)

Additional information:

This PR restructures the OpenShift Dedicated documentation to align with the ROSA restructure #41923/#43807.

### CODE REVIEW

File: _topic_maps/_topic_map_osd.yml [hunk 1]
```diff
@@ -32,16 +32,42 @@ Topics:
...
+Name: Introduction to OpenShift Dedicated
 Dir: osd_architecture
 Distros: openshift-dedicated
 Topics:
-- Name: Introduction to OpenShift Dedicated
+- Name: Understanding OpenShift Dedicated
   File: osd-understanding
 - Name: Architecture concepts
   File: osd-architecture
+- Name: Understanding your cloud deployment options
```
Comment (pneedle-rh) [inline _topic_maps/_topic_map_osd.yml:43]: This item could potentially go in the "Getting started" book and after the restructure we could build on it to make it similar to https://docs.openshift.com/rosa/rosa_getting_started/rosa-sts-getting-started-workflow.html in the ROSA library.

File: _topic_maps/_topic_map_osd.yml [hunk 2]
```diff
@@ -51,45 +77,69 @@ Topics:
...
 Distros: openshift-dedicated
 Topics:
 - Name: Getting started with OpenShift Dedicated
-  File: osd-quickstart
+  File: osd-getting-started
 ---
-Name: Red Hat OpenShift Cluster Manager
-Dir: ocm
+Name: Installing, accessing, and deleting OpenShift Dedicated clusters
+Dir: osd_install_access_delete_cluster
```
Comment (pneedle-rh) [inline _topic_maps/_topic_map_osd.yml:87]: I noticed that we don't have pages on granting/revoking privileges and access or deleting an OSD cluster in the "Installing, accessing, and deleting OpenShift Dedicated clusters". Maybe we can reuse the modules that are in the OSD "Getting started" page (https://docs.openshift.com/dedicated/osd_quickstart/osd-quickstart.html) in an assembly on that topic for the time being in this section and then build it out with more detail that is provided in the "Getting started" at a later date.

This is something we can look at after the restructure.

**Review by pneedle-rh — COMMENTED** [review]
@EricPonvelle this new library structure looks great, this is a fantastic job. The consistency with the ROSA library is really helpful for findability too.

I have added a couple of suggestions for reference in this PR directly, although they might be out of scope for the restructure and instead we could work on them later as a team.

**Review by lpettyjo — COMMENTED** [review]
Spot checked links and LGTM!

**Pull request merged**